spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -158,61 +168,60 @@ public void onSuccess(String result) {
 
 	@Override
 	public Subscriber subscribe(String subscription, MessageReceiver messageHandler) {
-		Subscriber subscriber = this.subscriberFactory.createSubscriber(subscription, messageHandler);
+		Subscriber subscriber =
+				this.subscriberFactory.createSubscriber(subscription, messageHandler);
 		subscriber.startAsync();
 		return subscriber;
 	}
 
 	/**
 	 * Pulls messages synchronously, on demand, using the pull request in argument.
-	 *
-	 * <p>This method acknowledges all received messages.
 	 * @param pullRequest pull request containing the subscription name
-	 * @return the list of {@link PubsubMessage} containing the headers and payload
+	 * @return the list of {@link AcknowledgeablePubsubMessage} containing the ack ID, subscription
+	 * and acknowledger
 	 */
-	private List<PubsubMessage> pull(PullRequest pullRequest, RetrySettings retrySettings) {
+	private List<AcknowledgeablePubsubMessage> pull(PullRequest pullRequest) {
 		Assert.notNull(pullRequest, "The pull request cannot be null.");
 
-		try {
-			SubscriberStub subscriber = this.subscriberFactory.createSubscriberStub(retrySettings);
-			Assert.notNull(subscriber, "A SubscriberStub is needed to execute the pull request.");
-
-			PullResponse pullResponse =	subscriber.pullCallable().call(pullRequest);
-
-			// Ack received messages.
-			if (pullResponse.getReceivedMessagesCount() > 0) {
-				List<String> ackIds = pullResponse.getReceivedMessagesList().stream()
-						.map(ReceivedMessage::getAckId)
+		PullResponse pullResponse =	this.subscriberStub.pullCallable().call(pullRequest);
+		List<AcknowledgeablePubsubMessage> receivedMessages =
+				pullResponse.getReceivedMessagesList().stream()
+						.map(message -> {
+							return new AcknowledgeablePubsubMessage(message.getMessage(),
+									message.getAckId(),
```

Comment:
I think we don't need this separate property. The getter can just delegate to the provided `PubsubMessage`.

Reply:
`PubsubMessage` doesn't contain an ack ID, `ReceivedMessage` does.
But `ReceivedMessage` doesn't contain the subscription name, so we'll always have to add it to a wrapper.
And since we have to do it for the subscription name, might as well do it for the ack ID too and thus we don't need `ReceivedMessage` anymore.

Reply:
Yep! I see now. Missed the fact that you use `message.getMessage()`.

I'm fine with what you have so far here.

Sorry for the noise!